### PR TITLE
json2outputs in run

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25728

## Description
Add a `json_to_outputs` flag in `run` command in order to use the raw_output and make a yaml output.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/run_cmd/runner.py [hunk 1]
```diff
@@ -48,12 +55,29 @@ def run(self):
...
+                with tempfile.NamedTemporaryFile(mode='w+') as f:
+                    if isinstance(raw_output_json, dict):
+                        f.write(json.dumps(raw_output_json))
+                    if isinstance(raw_output_json, list):
+                        f.write(json.dumps(raw_output_json[0]))
+                    f.seek(0)
+                    file_path = f.name
+                    command = self.query.split(' ')[0]
+                    print("What would you like the base path for the outputs that the script generates to be?")
+                    prefix = input()
```
Comment (bakatzir) [inline demisto_sdk/commands/run_cmd/runner.py:78]: we usually shy away from giving inputs in the middle of a demisto-sdk command run, this way we can use them as part of a different automation, just as you did with `json_to_outputs`.
put it as an arg, and think of a default path.

File: demisto_sdk/commands/run_cmd/runner.py [hunk 2]
```diff
@@ -136,6 +160,39 @@ def _export_debug_log(self, log_ids: list):
...
+                        for line in log_info:
+                            if self.RAW_RESPONSE_HEADER.match(line):
+                                temp_line = log_info.readline()
+                                temp_dict = json.loads(temp_line)
+                                output_file.write(temp_line.encode('utf-8'))
+                            else:
+                                output_file.write(line.encode('utf-8'))
+            print_color(f'Debug Log successfully exported to {self.debug_path}', LOG_COLORS.GREEN)
+            return temp_dict
+        return dict()
```
Comment (bakatzir) [inline demisto_sdk/commands/run_cmd/runner.py:194]: is this line reachable?

File: demisto_sdk/__main__.py [hunk 3]
```diff
@@ -540,7 +545,7 @@ def run(**kwargs):
 )
 @click.option(
     '--url', '-u',
-    help='URL to a Demisto instance. You can also specify the URL as an environment variable named: DEMISTO_BASE_URL'
+    help='URL to a Demisto instance. You can also speciFy the URL as an environment variable named: DEMISTO_BASE_URL'
```
Comment (bakatzir) [inline demisto_sdk/__main__.py:548]:
```suggestion
    help='URL to a Demisto instance. You can also specify the URL as an environment variable named: DEMISTO_BASE_URL'
```

**Comment by orlichter1** [comment]
@yaakovi could you have a look as well? 😄

File: demisto_sdk/commands/run_cmd/runner.py [hunk 4]
```diff
@@ -54,6 +61,24 @@ def run(self):
             else:
                 self._export_debug_log(log_ids)
 
+        if self.json2outputs:
+            if not self.prefix:
+                print_error("A prefix for the outputs is needed for this command. Please provide one")
```
Comment (yaakovi) [inline demisto_sdk/commands/run_cmd/runner.py:66]: this should fail the command.

File: demisto_sdk/commands/run_cmd/runner.py [hunk 5]
```diff
@@ -54,6 +61,24 @@ def run(self):
...
+                        if isinstance(raw_output_json, dict):
+                            f.write(json.dumps(raw_output_json))
+                        if isinstance(raw_output_json, list):
+                            f.write(json.dumps(raw_output_json[0]))
+                        f.seek(0)
+                        file_path = f.name
+                        command = self.query.split(' ')[0]
+                        json_to_outputs(command, file_path, self.prefix)
+                else:
+                    print_error("Could not extract raw output as JSON from command")
```
Comment (yaakovi) [inline demisto_sdk/commands/run_cmd/runner.py:80]: this should fail the command

File: demisto_sdk/commands/run_cmd/runner.py [hunk 6]
```diff
@@ -136,6 +161,44 @@ def _export_debug_log(self, log_ids: list):
...
+    def _return_raw_outputs_from_log(self, log_ids: list) -> dict:
+        """
+            retrieves the raw_outputs section from the debug_log
+        Args:
+            log_ids (list): artifact ids of the log files
+
+        Returns:
+            the raw_outputs of the executed query
+        """
+        if not self.debug_path:
```
Comment (yaakovi) [inline demisto_sdk/commands/run_cmd/runner.py:173]: when you have an `else` statement, there is no reason to use negative conditions (for future reference)

File: demisto_sdk/commands/run_cmd/runner.py [hunk 7]
```diff
@@ -24,17 +28,20 @@ class Runner:
     ERROR_ENTRY_TYPE = 4
     DEBUG_FILE_ENTRY_TYPE = 16
     SECTIONS_HEADER_REGEX = re.compile(r'^(Context Outputs|Human Readable section|Raw Response section)')
+    RAW_RESPONSE_HEADER = re.compile(r'^Raw Response section')
```
Comment (yaakovi) [inline demisto_sdk/commands/run_cmd/runner.py:31]: do we want to try extracting the data from a `Context Outputs` section first? as this is a better representation of the command output (which may differ from the raw response)?
if not available (or empty) we can default to the `Raw Response` section.

File: demisto_sdk/commands/run_cmd/runner.py (hunk 6 above)
Comment (orlichter1) [reply demisto_sdk/commands/run_cmd/runner.py:173]: What do you mean? put the `else` part first in this case?

File: demisto_sdk/commands/run_cmd/runner.py (hunk 7 above)
Comment (orlichter1) [reply demisto_sdk/commands/run_cmd/runner.py:31]: @ronykoz What do you think? As you were the one who specified this task.

Comment (orlichter1) [reply demisto_sdk/commands/run_cmd/runner.py:31]: Will do

File: demisto_sdk/commands/run_cmd/runner.py (hunk 6 above)
Comment (yaakovi) [reply demisto_sdk/commands/run_cmd/runner.py:179]: yes

**Review by yaakovi — APPROVED** [review]

**Comment by orhovy** [comment]
@orlichter1  Can you add changelog

**Pull request merged**